### PR TITLE
Code scripts for Problem 1 and Problem 4

### PR DESCRIPTION
**Pull request by RohanKarthikeyan**
@sbarat321 Please review the solution. Especially, make sure that:
- the assumptions stated are right or not; and
- if there any assumptions I forgot to add, please inform.

### CODE REVIEW

**Review by amondalgit — CHANGES_REQUESTED** [review]
A few changes suggested,

1. Add the problem statements originally given in **Assignment_02** pdf, before each part of the sub-problems (Optional).
2. Add a solution of **Part D** (After `line 141`)
```
NLL_Model4 <- function(theta, y, X)
{
  beta_0 = theta[1]
  beta_1 = theta[2]
  sigma = exp(theta[3])  
  
  n = length(y)
  l = 0
  for (i in 1:n)
  {
    if (y[i] != 0)
      {
      l = l + dgamma(y[i], exp(beta_0 + beta_1*log(X[i])),scale = sigma, log=T)
    }
  }
  return(-l)
}

theta_initial4 = c(1, 0.15, 0.5)
fit_4 = optim(theta_initial4, NLL_Model4,
              y=Insurance$Claims,
              X=Insurance$Holders,
              control=list(maxit=1500))
theta_hat = fit_4$par
beta0_hat = theta_hat[1]
beta1_hat = theta_hat[2]
sigma_hat = exp(theta_hat[3])
paste0("Estimated beta0: ", beta0_hat)
paste0("Estimated beta1: ", beta1_hat)
paste0("Estimated sigma: ", sigma_hat)

plot(log(Insurance$Holders), log(Insurance$Claims), main = "'Claims' ~ Gamma(alpha, sigma)", ylab = "Log(Claims)", xlab = "Log(Holders)")
abline(beta0_hat, beta1_hat)
```
3. Add a **bar graph** plotting all **BIC** values we got from different fits (After `line 168`)
4. A brief conclusion can be added at the end.

**Comment by amondalgit** [comment]
Reviewed Problem-4

File: Assignment_2/code/Problem-4.Rmd [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+## Model 3: Linear regression for log-normally distributed data
+
+```{r}
+NLL_Model3 <- function(theta, y, X)
+{
+  beta_0 = theta[1]
+  beta_1 = theta[2]
+  sigma = exp(theta[3])  # variance of error terms
+
+  n = length(data)
```
Comment (amondalgit) [inline Assignment_2/code/Problem-4.Rmd]: Please, change `data` to `y`, as `data` is a global variable

Comment (RohanKarthikeyan) [reply Assignment_2/code/Problem-4.Rmd]: OK.

**Review by amondalgit — APPROVED** [review]

**Pull request merged**